docs/data/joy/guides/overriding-component-structure/overriding-component-structure.md

```diff
@@ -0,0 +1,114 @@
+# Overriding component structure
+
+<p class="description">Learn how to override the default DOM structure of Joy UI components.</p>
+
+Joy UI components are designed to suit the widest possible range of use cases, but you may occasionally need to change how a component's structure is rendered in the DOM.
+
+To understand how to do this, it helps to have an accurate mental model of MUI components.
+
+## The mental model
+
+A component's structure is determined by the elements that fill that component's **slots**.
+Slots are most commonly filled by HTML tags, but may also be filled by React components.
+
+All components contain a root slot that defines their primary node in the DOM tree; more complex components also contain additional interior slots named after the elements they represent.
+
+All _non-utility_ Joy UI components accept two props for overriding their rendered HTML structure:
+
+- `component`—to override the root slot
+- `slots`—to override any interior slots (when present) as well as the root
+
+Additionally, you can pass custom props to interior slots using `slotProps`.
+
+## The root slot
+
+The root slot represents the component's "primary" element.
```

Comment:
```suggestion
The root slot represents the component's "outer-most" element.
```
I think using `primary` could be misleading because in some component the root slot is not a primary element, it is just a container for styling purpose.

For example, the `Input` component's primary element is `input` (maybe my perception of the word `primary` is different).

Reply:
That's a good point!